service/service.hpp

```diff
@@ -352,6 +352,46 @@ namespace raven
         send_answer(sock, answer);
     }
 
+    void get_settings_names(json::json &json_data, uvw::PipeHandle &sock)
+    {
+        LOG_SCOPE_F(INFO, __PRETTY_FUNCTION__);
+        auto cfg = fill_request<config_get_settings_names>(json_data);
+        DLOG_F(INFO, "cfg.id: %lu", cfg.id.value());
+        if (!config_clients_registry_.at(sock.fileno()).has_loaded(raven::config_id_st{cfg.id})) {
+            send_answer(sock, request_state::unknown_id);
+            return ;
+        }
+        auto config_json_data = db_.get_config(config_clients_registry_.at(sock.fileno()).get_db_id_from(raven::config_id_st{cfg.id}));
+        if (db_.fail()) {
+            send_answer(sock, request_state::db_error);
+            return ;
+        }
+        json::json settings_name;
+        for (auto &[key, value] : config_json_data[config_settings_field_keyword].items()) {
+            settings_name.push_back(key);
+        }
+        config_get_settings_names_answer answer{std::move(settings_name), convert_request_state.at(request_state::success)};
+        send_answer(sock, answer);
+    }
+
+    void get_settings_all(json::json &json_data, uvw::PipeHandle &sock)
```

Comment:
I don't really like this function name.

Reply:
I don't like calling it get_settings because, in the future get_setting will be able to take a list of settings to get and not a single one, this name shows that given the id it will return all settings within the corresponding config

Reply:
btw I'm also for changing the names of every request that take a list of settings as parametter from blabla_setting to blabla_settings or even propose both in the communication protocol (so in those case get_settings will be reserved by the function that take a list of settings as parametter)

Reply:
`get_all_settings` would at least be proper English

Reply:
Agree, I chose get_settings_all to specify first that this is a get_settings function and then that it fetches every settings, but yeah it's easier to read with the proper english one